### PR TITLE
DDF-1366 Use framework to get source status in AdminPollerServiceBean

### PR DESCRIPTION
**Pull request by kcover**
####DO NOT MERGE BEFORE THE NEW SOURCEPOLLER IMPLEMENTATION

#### What does this PR do?
Modifies the AdminPollerServiceBean to use the CatalogFramework to
check the availability of the sources and adds tests for the new
functionality.

#### Who is reviewing it? 

@emmberk @peterhuffer 

#### Select relevant component teams: 

@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@clockard 
@vinamartin

#### How should this be tested?

build and install ddf. Setup some sources and make sure the sources tab in the catalog app is updated when the status of a source changes. Make sure that the source isn't queried each time the refresh button in the tab is clicked.

#### What are the relevant tickets?

[DDF-1366](https://codice.atlassian.net/browse/DDF-1366)

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java [hunk 1]
```diff
@@ -147,11 +156,8 @@ public boolean sourceStatus(String servicePID) {
           try {
             Configuration config = helper.getConfiguration(cs);
             if (config != null && config.getProperties().get("service.pid").equals(servicePID)) {
-              try {
-                return source.isAvailable();
-              } catch (Exception e) {
-                LOGGER.info("Couldn't get availability on source {}: {}", servicePID, e);
-              }
+                final String sourceId = source.getId();
+                return sourceIsAvailable(sourceId);
```
Comment (clockard) [inline catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: Could we just make this `sourceIsAvailable(source.getId());`

Comment (paouelle) [reply catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: ... or pull it up before the try so you can reuse that local variable in the log at line 163

File: catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java [hunk 2]
```diff
@@ -145,12 +165,57 @@ public void testAllSourceInfo() {
...
+    assertThat(poller.sourceStatus(CONFIG_PID), is(false));
+  }
+
+  @Test
+  public void testSourceStatusWithUnknownPID(){
+    assertThat(poller.sourceStatus("UnknownPID"), is(false));
+  }
+
+  @Test
+  public void testSourceStatusWithNonConfiguredService()throws Exception{
```
Comment (emmberk) [inline catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java]: Looks like these 2 files will fail the formatting check

File: catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java [hunk 3]
```diff
@@ -167,6 +173,22 @@ public boolean sourceStatus(String servicePID) {
...
+      final SourceInfoRequest sourceInfoRequest =
+          new SourceInfoRequestSources(false, Collections.singleton(sourceId));
+      final SourceInfoResponse sourceInfoResponse =
+          catalogFramework.getSourceInfo(sourceInfoRequest);
+      final Set<SourceDescriptor> sourceDescriptors = sourceInfoResponse.getSourceInfo();
+      final SourceDescriptor sourceDescriptor =
+          Iterables.getOnlyElement(sourceDescriptors);
+      return sourceDescriptor.isAvailable();
+    } catch (Exception e){
+      LOGGER.info("Couldn't get availability on source {}: {}", sourceId, e);
```
Comment (emmberk) [inline catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: ✏️ `LOGGER.info("Couldn't get availability of source {}", sourceId, e);`

File: catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java [hunk 4]
```diff
@@ -145,10 +160,57 @@ public void testAllSourceInfo() {
   }
 
   @Test
-  public void testSourceStatus() {
+  public void testSourceStatusAvailable() throws Exception {
+    SourceDescriptorImpl Descriptor = mock(SourceDescriptorImpl.class);
```
Comment (peterhuffer) [inline catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java]: ✏️ `Descriptor` -> `descriptor`.

Comment (peterhuffer) [inline catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java]: ❓ Why do we mock the `SourceDescriptorImpl` here and not the `SourceDescriptor`? Should mock the interface if we can.

File: catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java [hunk 5]
```diff
@@ -145,10 +160,57 @@ public void testAllSourceInfo() {
...
+    when(catalogFramework.getSourceInfo(any(SourceInfoRequest.class))).thenReturn(Response);
+
     assertThat(poller.sourceStatus(CONFIG_PID), is(true));
-    assertThat(poller.sourceStatus(EXCEPTION_PID), is(false));
-    assertThat(poller.sourceStatus("FAKE SOURCE"), is(false));
+  }
+
+  @Test
+  public void testSourceStatusUnavailable() throws Exception {
+    SourceDescriptorImpl Descriptor = mock(SourceDescriptorImpl.class);
```
Comment (peterhuffer) [inline catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java]: ❓ ✏️ Same 2 comments as above.

File: catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java [hunk 6]
```diff
@@ -145,10 +160,57 @@ public void testAllSourceInfo() {
   }
```
Comment (peterhuffer) [inline catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java]: ✏️ We should have a test where multiple SourceDescriptors are returned and the RuntimeException is thrown from `Iterables.getOnlyElement`.

File: catalog/admin/admin-poller-service/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 7]
```diff
@@ -21,6 +21,8 @@
 
     <reference id="configurationAdmin" interface="org.codice.ddf.admin.core.api.ConfigurationAdmin"/>
 
+    <reference id="catalogFramework" interface="ddf.catalog.CatalogFramework" availability="mandatory"/>
```
[the remark this answers is not in the document]
Comment (kcover) [reply catalog/admin/admin-poller-service/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: I'm not quite following. How does it give us a proxy? Could you explain in more detail please?

File: catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java (hunk 6 above)
Comment (kcover) [reply catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBeanTest.java]: Done.

**Pull request closed without merging**